### PR TITLE
[CSL-1563] Use TH to fetch/print git revision

### PR DESCRIPTION
**Pull request by volhovM**
This PR provides a simple TH-based framework (`Pos.Util.CompileInfo`) to fetch the compile time data like git revision. Previous solution with cpp definition led to unneeded recompilations and now this problem is solved.

### CODE REVIEW

**Comment by gromakovsky** [comment]
Also there was a request to print revision when `--version` is passed.

File: CODEOWNERS [hunk 1]
```diff
@@ -41,13 +41,13 @@ core/Pos/Merkle.hs              @neongreen
 
 core/Pos/Util/Config.hs         @neongreen
 core/Pos/Util/Config/           @neongreen
-node/src/Pos/CompileConfig/     @neongreen
+lib/src/Pos/CompileConfig/     @neongreen
```
Comment (gromakovsky) [inline CODEOWNERS]: Indentation :(

File: scripts/build/cardano-sl.sh [hunk 2]
```diff
@@ -174,6 +174,8 @@ do
     spec_prj="wallet"
   elif [[ $var == "explorer" ]]; then
     spec_prj="explorer"
+  elif [[ $var == "node" ]]; then
+    spec_prj="node"
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: You should also modify `sl+` case to build `node`.

File: auxx/src/Command/Tx.hs [hunk 3]
```diff
@@ -215,6 +218,7 @@ sendTxsFromFile
        , HasUpdateConfiguration
        , HasNodeConfiguration
        , HasGtConfiguration
+       , HasCompileInfo
```
Comment (gromakovsky) [inline auxx/src/Command/Tx.hs]: It's strange that it's needed here.

File: lib/README.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+<!-- THIS IS AUTOGENERATED FILE. DO NOT CHANGE IT MANUALLY! -->
```
Comment (gromakovsky) [inline lib/README.md]: How is this file generated? Will it be correctly auto generated after being moved to `lib/`?

Comment (volhovM) [reply lib/README.md]: cc @denisshevchenko

**Pull request merged**